src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php

```diff
@@ -1227,6 +1252,71 @@ public function findServiceGroupsByHostAndService(int $hostId, int $serviceId):
         return $serviceGroups;
     }
 
+    /**
+     * Find downtimes for host or service
+     * @param int $hostId
+     * @param int $serviceId
+     * @return array
+     */
+    public function findDowntimes(int $hostId, int $serviceId): array
+    {
+        $downtimes = [];
+
+        if (empty($hostId)) {
+            return $downtimes;
+        }
+
+        $sql = 'SELECT * FROM `:dbstg`.`downtimes` WHERE host_id = :hostId AND service_id = :serviceId ' .
+                'AND deletion_time IS NULL ORDER BY entry_time DESC LIMIT 500';
```

Comment:
current date should be greater than `entry_time`
If you don't do this check, you will get downtimes which are planned in the future
Furthermore, IMHO, we should remove `LIMIT 500` here, and just get the list we get from query (which will have a few results cause it will be only current downtimes)

Reply:
thanks, I will push a fix for this after clarifying the Business Requirements with PO